### ISSUE TITLE
[autocomplete] accessibility: Command palette example keyboard shortcuts not announced by screen readers

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4248]
# Bug report

## Current behavior

On opening the command palette dialog, there are some keyboard shortcuts shown at the bottom. These are not announced by screen readers and wouldn't be unless they switched from forms to reading mode (which is unlikely when the dialog only contains an autocomplete combobox). 

People using a screen reader will almost certainly miss these keyboard shortcuts. 

This could justifiably count as a fail against [Understanding WCAG 2.2 Success Criterion 1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) or possibly [Understanding WCAG 2.2 Success Criterion 3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html) as this content is potentially important/helpful information that is not referenced by the autocomplete.

## Expected behavior

Keyboard commands should be announced to screen reader users. This could be done by referencing them via `aria-describedby` on the input or by placing them inside a named focusable region 

## Reproducible example

Steps to reproduce: 
1. Turn on VoiceOver on a Mac (or other screen reader
2. Navigate to the [Autocomplete: Command Palette example](https://base-ui.com/react/components/autocomplete#command-palette)
3. Activate the "Open command palette" button
4. Move around the dialog content. (e.g. using Tab, Shift+Tab and Arrow keys)
5. Note the keyboard commands are not announced 

## Base UI version

v1.2.0

## Which browser are you using?

Chrome

## Which OS are you using?

Mac OS

## Which assistive tech are you using (if applicable)?

Voiceover

## Additional context

Note: The text also instructs users to press `cmd+k` when Windows users would need to use `Ctrl`. The content may need to reflect this.
